ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -247,26 +240,26 @@ Control {
                 onSearchTextChanged: d.searchText = searchText
             }
 
-            Item { Layout.fillHeight: !itemTag.visible }
+            Item { Layout.fillHeight: !maxSendButton.visible }
 
-            StatusListItemTag {
-                id: itemTag
-                objectName: "maxTagButton"
+            MaxSendButton {
+                id: maxSendButton
                 Layout.alignment: Qt.AlignRight
                 Layout.maximumWidth: parent.width
-                Layout.preferredHeight: 22
+                objectName: "maxTagButton"
+
+                value: d.maxInputBalance
+                symbol: d.inputSymbol
+                valid: amountToSendInput.input.valid || !amountToSendInput.input.text
+                valueIsFiat: amountToSendInput.inputIsFiat
+                formatCurrencyAmount: (amount, symbol, options) => root.currencyStore.formatCurrencyAmount(amount, symbol, options)
+
                 visible: d.isSelectedHoldingValidAsset && root.swapSide === SwapInputPanel.SwapSide.Pay
-                title: d.maxInputBalance > 0 ? qsTr("Max: %1").arg(d.maxInputBalanceFormatted)
-                                             : qsTr("No balances active")
-                tagClickable: true
-                closeButtonVisible: false
-                titleText.font.pixelSize: 12
-                bgColor: amountToSendInput.input.valid || !amountToSendInput.input.text ? Theme.palette.primaryColor3 : Theme.palette.dangerColor2
-                titleText.color: amountToSendInput.input.valid || !amountToSendInput.input.text ? Theme.palette.primaryColor1 : Theme.palette.dangerColor1
-                onTagClicked: {
-                    const max = d.prepareForMaxSend(d.maxInputBalance, d.inputSymbol)
+
+                onClicked: {
+                    const max = maxSafeValue
```

Comment:
doesnt need to be created as const anymore? can be used directly?

Reply:
```qml
const max = maxSafeValue
if (max > 0)
  amountToSendInput.input.text = maxSafeValueAsString
```

That's a numeric value used for the check; below we use the string/text value to insert into the input field

Reply:
Oh you mean changing it to this:
```qml
if (maxSafeValue > 0)
```
?

Reply:
Got it, ammended :)